CMakeModules/CompilerSettings.cmake

```diff
@@ -27,12 +27,15 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
 endif()
 
 # Visual Studio
-if(MSVC OR MSVC90 OR MSVC10)
+if(MSVC)
+    if(MSVC_VERSION VERSION_LESS 1900)
+        message(FATAL_ERROR "${PROJECT_NAME} requires Visual Studio 2015 or greater.")
+    endif()
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /EHsc /MP /W1 /bigobj")
     if(FCL_TREAT_WARNINGS_AS_ERRORS)
         add_definitions(/WX)
     endif(FCL_TREAT_WARNINGS_AS_ERRORS)
```

Comment:
I agree! `endif()` is always better, IMO. That is very clear when the original condition is some complicated expression!